src/meta/api/src/schema_api_impl.rs

```diff
@@ -1075,6 +1080,36 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
         })
     }
 
+    #[tracing::instrument(level = "debug", ret, err, skip_all)]
+    async fn update_index(&self, req: UpdateIndexReq) -> Result<UpdateIndexReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let index_id_key = IndexId {
+            index_id: req.index_id,
+        };
+
+        let (index_meta_seq, _) = get_index_meta_by_id_or_err(self, &index_id_key).await?;
+
+        if index_meta_seq == 0 {
+            return Err(KVAppError::AppError(AppError::UnknownIndex(
+                UnknownIndex::new(&req.index_name, "update_index"),
+            )));
+        }
```

Comment:
Consider checking the reply at line 1099 after executing `upsert_kv` function to decide if to return the `UnknownIndex` error.

Save one meta-service RPC.

And `get_index_meta_by_id_or_err` can be removed